scanner/k8s-assets/processor/processor.go

```diff
@@ -0,0 +1,179 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package processor
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/cloudoperators/heureka/scanners/k8s-assets/client"
+	"github.com/cloudoperators/heureka/scanners/k8s-assets/scanner"
+	log "github.com/sirupsen/logrus"
+)
+
+type Processor struct {
+	Client *graphql.Client
+	config Config
+}
+
+type CCRN struct {
+	Region    string
+	Domain    string
+	Project   string
+	Cluster   string
+	Namespace string
+	Pod       string
+	PodID     string
+	Container string
+}
+
+func (c CCRN) String() string {
+	// Define default CCRN
+	return fmt.Sprintf("rn.cloud.sap/ccrn/kubernetes/v1/%s/%s/%s/kubernikus/%s/%s/%s/%s/%s",
+		c.Region,
+		c.Domain,
+		c.Project,
+		c.Cluster,
+		c.Namespace,
+		c.Pod,
+		c.PodID,
+		c.Container,
+	)
+}
+
+func NewProcessor(cfg Config) *Processor {
+	httpClient := http.Client{}
+	gClient := graphql.NewClient(cfg.HeurekaUrl, &httpClient)
+	return &Processor{
+		config: cfg,
+		Client: &gClient,
+	}
+}
+
+// ProcessService creates a service and processes all its pods
+func (p *Processor) ProcessService(ctx context.Context, namespace string, serviceInfo scanner.ServiceInfo) (string, error) {
+	var serviceId string
+
+	// Create new Service
+	createServiceInput := &client.ServiceInput{
+		Name: serviceInfo.Name,
+	}
+
+	createServiceResp, err := client.CreateService(ctx, *p.Client, createServiceInput)
+	if err != nil {
+		// The Service might already exist in the DB
+		// Let's try to fetch one Service by name
+		_serviceId, err := p.getService(ctx, serviceInfo)
+		if err != nil {
+			return "", fmt.Errorf("failed to create Service %s: %w", serviceInfo.Name, err)
+		}
+		serviceId = _serviceId
+	} else {
+		serviceId = createServiceResp.CreateService.Id
+	}
```

Comment:
Why don't we do it the other way around, first checking if it already exists and then creating it if it doesn't?